plugins/modules/jenkins_credential.py

```diff
@@ -0,0 +1,831 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+---
+module: jenkins_credential
+short_description: Manage Jenkins credentials and domains via API
+version_added: 1.0.0
+description:
+  - This module allows managing Jenkins credentials and domain scopes via Jenkins HTTP API.
+  - You can create, update, and delete different credential types such as username/password, secret text, SSH key, certificates, GitHub App, and scoped domains.
+  - For scoped credentials, it supports hostname, hostname:port, path, and scheme-based restrictions.
+requirements:
+  - requests
+  - python >= 3.6
+author:
+  - Youssef Ali (@YoussefKhalidAli)
+extends_documentation_fragment:
+  - community.general.attributes
+attributes:
+  check_mode:
+    support: full
+    details:
+      - Module fully supports check mode and will report changes without making them
+  diff_mode:
+    support: none
+options:
+  id:
+    description:
+      - The ID the Jenkins credential or domain.
+    required: true
+    type: str
+  type:
+    description:
+      - Type of the credential or action.
+    choices:
+      - userAndPass
+      - file
+      - text
+      - githubApp
+      - sshKey
+      - certificate
+      - scope
+    type: str
+  command:
+    description:
+      - The operation to perform.
+    choices: [add, delete, update]
+    default: add
+    type: str
+  scope:
+    description:
+      - Jenkins credential domain scope.
+    type: str
+    default: '_'
+  url:
+    description:
+      - Jenkins server URL.
+    type: str
+    default: http://localhost:8080
+  jenkinsUser:
+    description:
+      - Jenkins user for authentication.
+    required: true
+    type: str
+  token:
+    description:
+      - Jenkins API token.
+    required: true
+    type: str
+  description:
+    description:
+      - Optional description of the credential or domain.
+    default: ''
+    type: str
+  username:
+    description:
+      - Username for credentials that require it (e.g., sshKey, userAndPass).
+    type: str
+  password:
+    description:
+      - Password or secret text.
+    type: str
+  secret:
+    description:
+      - Secret text (used in "text" type).
+    type: str
+  appID:
+    description:
+      - GitHub App ID.
+    type: str
+  owner:
+    description:
+      - GitHub App owner.
+    type: str
+  filePath:
```

Comment:
Generally you should use `snake_case` for options, not `camelCase`. It's OK to add a camel case alias if that's way more common for Jenkins, but snake case should be the preferred form.